src/watertap_contrib/seto/unit_models/surrogate/tests/test_vagmd_surrogate.py

```diff
@@ -0,0 +1,511 @@
+        m.fs.water_properties = WaterParameterBlock()
+
+        # System specification (Input variables)
+        feed_flow_rate = 600  # 400 - 1100 L/h
+        evap_inlet_temp = 80  # 60 - 80 deg C
+        cond_inlet_temp = 25  # 20 - 30 deg C
+        feed_temp = 25  # 20 - 30 deg C
+        feed_salinity = 35  # 35 - 292 g/L
+        initial_batch_volume = 50  # > 50 L
+        recovery_ratio = 0.5  # -
+        module_type = "AS7C1.5L"
+        cooling_system_type = "closed"
+        cooling_inlet_temp = (
+            25  # deg C, not required when cooling system type is "closed"
+        )
+
+        # Identify if the final brine salinity is larger than 175.3 g/L for module "AS7C1.5L"
+        # If yes, then operational parameters need to be fixed at a certain value,
+        # and coolying circuit is closed to maintain condenser inlet temperature constant
+        final_brine_salinity = feed_salinity / (1 - recovery_ratio)  # = 70 g/L
+        if module_type == "AS7C1.5L" and final_brine_salinity > 175.3:
+            cooling_system_type = "closed"
+            feed_flow_rate = 1100  # L/h
+            evap_inlet_temp = 80  # deg C
+            cond_inlet_temp = 25  # deg C
+            high_brine_salinity = True
+        else:
+            high_brine_salinity = False
+
+        m.fs.vagmd = VAGMDSurrogate(
+            property_package_seawater=m.fs.seawater_properties,
+            property_package_water=m.fs.water_properties,
+            module_type=module_type,
+            high_brine_salinity=high_brine_salinity,
+            cooling_system_type=cooling_system_type,
+        )
+
+        # Specify feed flow state properties
+        m.fs.vagmd.feed_props.calculate_state(
+            var_args={
+                ("flow_vol_phase", "Liq"): pyunits.convert(
+                    feed_flow_rate * pyunits.L / pyunits.h,
+                    to_units=pyunits.m**3 / pyunits.s,
+                ),
+                ("conc_mass_phase_comp", ("Liq", "TDS")): feed_salinity,
+                ("temperature", None): feed_temp + 273.15,
+                # feed flow is at atmospheric pressure
+                ("pressure", None): 101325,
+            },
+            hold_state=True,
+        )
+
+        # Specify evaporator inlet temperature
+        m.fs.vagmd.evaporator_in_props[0].temperature.fix(evap_inlet_temp + 273.15)
+
+        # Identify cooling system type
+        # Closed circuit, in which TCI is forced to be constant and the cooling water temperature can be adjusted.
+        if cooling_system_type == "closed":
+            m.fs.vagmd.condenser_in_props[0].temperature.fix(cond_inlet_temp + 273.15)
+        # Open circuit, in which cooling is available at a constant water temperature and condenser inlet temperature varies.
+        else:  # "open"
+            m.fs.vagmd.cooling_in_props[0].temperature.fix(cooling_inlet_temp + 273.15)
+
+        return m
+
+    @pytest.mark.unit
```

Comment:
I might've missed this because reviewing from phone, but do you have a test for assert_units_consistent()?

Reply:
It will fail for surrogate equations. Same in LT-MED and MED-TVC.